PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/RuleConfigLoader.java

```diff
@@ -189,4 +193,21 @@ private List<String> getListFromJsonArray(JsonReader reader) throws IOException
         }
         return list;
     }
+
+    private Map<String, String> getMappedItemsFromJsonReader(JsonReader reader, Map <String, String> mappedItemsToCheck) throws IOException {
+
+        // Check if the JSON file starts with an object
+        if (reader.nextToken() != JsonToken.START_OBJECT) {
+            throw new IOException("Expected start of object");
+        }
+
+        // Read the JSON file and parse the RuleConfig object
+        while (reader.nextToken() != JsonToken.END_OBJECT) {
+            String key = reader.getFieldName();
+            String value = reader.getString();
```

Comment:
You'll need to move the `JsonReader` to the `nextToken` before getting the `value`. Right now, this will have the `JsonReader` get the field name as a string.

Also, there are helper APIs on `JsonReader` that would simplify this.

```java
return reader.readMap(JsonReader::getString);
```

This helper method does the following:

- Validates that the `JsonReader` has been initialized (aka, currentToken != null) and is pointing to JsonToken.START_OBJECT
- Loops through the JSON object getting the field name and the value as a JsonToken.STRING

Reply:
for this, I ended up making a different method to parse the object and return a list since the implementation is using a List<Pattern> -- this way I can use it for other rules that are coming up too.